src/common/snippets/src/pass/extract_unsupported_transposes.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/extract_unsupported_transposes.hpp"
+
+#include "openvino/opsets/opset1.hpp"
+#include "snippets/pass/mha_tokenization.hpp"
+#include "snippets/itt.hpp"
+
+
+bool ov::snippets::pass::ExtractUnsupportedTransposes::run_on_subgraph(const std::shared_ptr<op::Subgraph>& subgraph) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::ExtractUnsupportedTransposes");
+    const auto& body = subgraph->body_ptr();
+    const auto parameters = body->get_parameters();
+    // [107806]: If count of Parameters isn't equal to Subgraph inputs,
+    //           we cannot guarantee correct extraction since we don't have correct connections between body I/O and Subgraph I/O.
+    OPENVINO_ASSERT(parameters.size() == subgraph->input_values().size(),
+                    "Failed to extract unsupported transposes: the count of Parameters isn't equal to Subgraph inputs");
+
+    bool updated = false;
+    for (size_t i = 0; i < parameters.size(); ++i) {
+        const auto& parameter = parameters[i];
+        const auto& consumers = parameter->get_output_target_inputs(0);
+        if (consumers.size() != 1)
+            continue;
+
+        const auto transpose = ov::as_type_ptr<opset1::Transpose>(consumers.begin()->get_node()->shared_from_this());
+        if (!transpose)
+            continue;
+
+        const auto& order = ov::as_type_ptr<opset1::Constant>(transpose->get_input_node_shared_ptr(1));
+        if (!order)
+            continue;
+
+        const auto order_value = order->cast_vector<int>();
```

Comment:
I am not sure that we can avoid copying here. All API, which can give us `std::vector` from ov::op::v0::Constant, creates the vector. I suppose we can avoid casting only (by using `get_vector` instead of `cast_vector`), but in this case we will have to handle i64 and i32 cases manually: I don't think that we really need it in case of transpose order vector, which is usually small enough

Reply:
I'm not suggesting to avoid casting, but rather asking why can't we use const reference here?

Reply:
Okay, now I understand your point. We can use const reference, but I think that thanks to Return Value Optimization we avoid additional copying anyway

Reply:
Fair enough, agreed.
Wow you guys hate references here 😅